### PR TITLE
Update etcd version to 3.1.11 for release 1.9

### PR DESCRIPTION
**Pull request by xiangpengzhao**
**What this PR does / why we need it**:
>This is fixing some bugs in HA setups.

ref: https://github.com/kubernetes/kubernetes/pull/57238
https://github.com/kubernetes/kubernetes/pull/57633#pullrequestreview-85582060 and discussions in that thread.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
ref: #57238

**Special notes for your reviewer**:
/cc @wojtek-t @luxas @jpbetz @mbohlool 

**Release note**:

```release-note
Update etcd version to 3.1.11
```

### CODE REVIEW

**Comment by xiangpengzhao** [comment]
bazel failed due to 
```
ERROR: /workspace/k8s.io/kubernetes/test/integration/framework/BUILD:8:1: no such package '@com_coreos_etcd//': Error downloading [https://github.com/coreos/etcd/releases/download/v3.1.11/etcd-v3.1.11-linux-amd64.tar.gz] to /root/.cache/bazel/_bazel_root/e9f728bbd90b3fba632eb31b20e1dacd/external/com_coreos_etcd/etcd-v3.1.11-linux-amd64.tar.gz: Checksum was b80c6c0719beee703821ece80f9e48b071ad5f59bb284bd3ea2104700738d9ec but wanted 2d335f298619c6fb02b1124773a56966e448ad9952b26fea52909da4fe80d2be and referenced by '//test/integration/framework:go_default_library'
```

@jpbetz any ideas?

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -442,47 +437,47 @@
 		},
 		{
 			"ImportPath": "github.com/containerd/containerd/api/services/containers/v1",
-			"Comment": "v1.0.0-beta.2-159-g27d450a0",
+			"Comment": "v1.0.0-beta.2-159-g27d450a",
```
Comment (wojtek-t) [inline Godeps/Godeps.json]: Why are we bumping all of these too? Are those really transitive dependencies of etcd client?

**Comment by wojtek-t** [comment]
@xiangpengzhao - thanks a lot for putting this together.
Did you have a chance to look into test failures? Those doesn't seem like flakes to me

File: Godeps/Godeps.json (hunk 1 above)
Comment (xiangpengzhao) [reply Godeps/Godeps.json]: I just followed the guide [Using godep to manage dependencies](https://github.com/kubernetes/community/blob/master/contributors/devel/godep.md) and it updated godeps for me. I'm not sure why this was updated. But maybe it's a mistake: `g27d450a0` vs `g27d450a`. I will revert it manually.

**Comment by xiangpengzhao** [comment]
>Did you have a chance to look into test failures? Those doesn't seem like flakes to me

The log shows the checksum of etcd-v3.1.11-linux-amd64.tar.gz is incorrect. @jpbetz do you know what may cause this failure? Thanks!
```
ERROR: /workspace/k8s.io/kubernetes/test/integration/framework/BUILD:8:1: no such package 
'@com_coreos_etcd//': Error downloading 
[https://github.com/coreos/etcd/releases/download/v3.1.11/etcd-v3.1.11-linux-amd64.tar.gz] to 
/root/.cache/bazel/_bazel_root/e9f728bbd90b3fba632eb31b20e1dacd/external/com_coreos_etcd/etcd-
v3.1.11-linux-amd64.tar.gz: Checksum was 
b80c6c0719beee703821ece80f9e48b071ad5f59bb284bd3ea2104700738d9ec but wanted 
2d335f298619c6fb02b1124773a56966e448ad9952b26fea52909da4fe80d2be and referenced by 
'//test/integration/framework:go_default_library'

```

File: Godeps/Godeps.json (hunk 1 above)
Comment (xiangpengzhao) [reply Godeps/Godeps.json]: Took another look and found that this is just a Comment change. I guess it's a upstream Comment change. Here it doesn't bump the real code.

File: Godeps/Godeps.json [hunk 2]
```diff
@@ -907,152 +907,152 @@
...
 		{
 			"ImportPath": "github.com/docker/distribution/reference",
-			"Comment": "v2.6.0-rc.1-209-gedc3ab29",
+			"Comment": "v2.6.0-rc.1-209-gedc3ab2",
 			"Rev": "edc3ab29cdff8694dd6feb85cfeb4b5f1b38ed9c"
 		},
 		{
 			"ImportPath": "github.com/docker/docker/api",
-			"Comment": "docs-v1.12.0-rc4-2016-07-15-7401-g4f3616fb1",
+			"Comment": "docs-v1.12.0-rc4-2016-07-15-7401-g4f3616f",
```
Comment (xiangpengzhao) [inline Godeps/Godeps.json]: we can seee the Godeps.json file in openshift. Its Comment is also `docs-v1.12.0-rc4-2016-07-15-7401-g4f3616f`.
https://github.com/openshift/origin/blob/master/Godeps/Godeps.json#L1219
```
		{
			"ImportPath": "github.com/docker/docker/api",
			"Comment": "docs-v1.12.0-rc4-2016-07-15-7401-g4f3616f",
			"Rev": "4f3616fb1c112e206b88cb7a9922bf49067a7756"
		},
```

File: Godeps/Godeps.json (hunk 1 above)
Comment (xiangpengzhao) [reply Godeps/Godeps.json]: See also: https://github.com/kubernetes/kubernetes/pull/57811#discussion_r159820267

**Comment by xiangpengzhao** [comment]
@wojtek-t @mbohlool all tests passed, PTAL. Thanks!

**Comment by xiangpengzhao** [comment]
@wojtek-t thanks!
ping @mbohlool for cherry-picking approval. Sorry for missing the 1.9.1 train...

**Comment by mbohlool** [comment]
Why didn't you use automatic tools to cherrypick #57238?

**Comment by wojtek-t** [comment]
@mbohlool - this is a different PR; that was just for build an image. This is bumping dependencies and using that image.
There wasn't corresponding PR in master branch, since we bumped etcd there directly to 3.2.12.

**Pull request merged**